tools/filters/join.py

```diff
@@ -294,9 +295,18 @@ def get_identifier_by_line( line, column, split=None ):
         return None
     if fill_options is None:
         fill_options = Bunch( fill_unjoined_only=True, file1_columns=None, file2_columns=None )
+    keep_headers_done = False
     out = open( out_filename, 'w+b' )
     index = BufferedIndex( filename2, column2, split, buffer, index_depth )
     for line1 in open( filename1, 'rb' ):
+        if keep_headers and not keep_headers_done:
```

Comment:
Isn't there a dataset attribute that tells you the number of header lines? This might be a good alternative to the regex? I don't know, I'd need someone else to comment on this.